shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/database/core/DataStore.kt

```diff
@@ -38,4 +39,11 @@ class DataStoreImpl(private val settings: Settings): DataStore {
     override fun <T> save(serializer: KSerializer<T>, key: String, value: T) {
         settings.encodeValue(serializer, key, value)
     }
+
+    override fun remove(key: String) {
+        settings
+            .keys
+            .filter { it.startsWith(key) }
+            .map { settings.remove(it) }
+    }
```

Comment:
Just curious, why don't we do exact string comparison here? with the `startsWith` function, we might remove more keys unexpectedly if multiple keys have the same prefixes and if this is intentional, I think we should reflect that into the function name also, for example:
```suggestion
    override fun removeStartingWith(key: String) {
        settings
            .keys
            .filter { it.startsWith(key) }
            .map { settings.remove(it) }
    }
```
 What do you think? 🤔

Reply:
The keys are mapped internally by the lib, this feature is in development from the lib so we have to add workaround. I will rename this to `removeObject(key:)`